primary_beam_correction.py

```diff
@@ -264,7 +247,7 @@ def main():
     data = read_fits(path)
     logging.info('----------------------------------------')
     logging.info('Getting the position of the phase centre')
-    phase_center, source_pos = get_positions(path)
+    phase_center, image_wcs = get_positions(path)
```

Comment:
Do you ever actually use image_wcs? Does `get_positions` need to return it?

Reply:
Yes, I need it. It is used on the radial_offset function.

Reply:
Yeah, but it seems to me you don;t need to call `get_positions` at all in main right.

Furthermore, you go through the whole process of getting `row` and `col` in `main` when that could just as easily be done inside `beam_pattern` since they aren't used anywhere else. Then you don't need to do `read_fits` inside `main` either.